### PR TITLE
[invalid][world rebuild] Check that GNUInstallDirs is used on Windows

### PR DESCRIPTION
**Pull request by Thomas1664**
This PR tries to detect if some ports use `CMAKE_INSTALL_MANDIR` without `GNUInstallDirs`. As a result man pages are installed to system root or the user gets a permission denied error.

### CODE REVIEW

File: scripts/cmake/vcpkg_common_definitions.cmake [hunk 1]
```diff
@@ -186,4 +186,6 @@ if(VCPKG_TARGET_IS_WINDOWS)
     list(APPEND VCPKG_SYSTEM_LIBRARIES Ws2_32)
     list(APPEND VCPKG_SYSTEM_LIBRARIES wldap32)
     list(APPEND VCPKG_SYSTEM_LIBRARIES crypt32)
+    
+    set(CMAKE_INSTALL_MANDIR "S:/")
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_common_definitions.cmake:190]: why would this be used be the CMakeLists.txt of CMake based ports if it is defined within the vcpkg script execution? Is it forwarded to CMake somewhere?

**Comment by BillyONeal** [comment]
It doesn't seem like this will do anything because you're applying it to `portfile.cmake`, not the actual build system invocations you're looking for.

**Comment by Thomas1664** [comment]
@BillyONeal Now I'm forwarding `CMAKE_INSTALL_MANDIR` to CMake

**Comment by dg0yt** [comment]
> Can you please cancel Linux, OSX and UWP to prevent wasting CI time?

Next time just add a guarded `FATAL_ERROR`.

**Comment by Thomas1664** [comment]
> > Can you please cancel Linux, OSX and UWP to prevent wasting CI time?
> 
> Next time just add a guarded `FATAL_ERROR`.

I don't see how a fatal error would help

**Comment by Thomas1664** [comment]
This didn't have the expected result: every port that installs man fails instead of only those without GNUInstallDirs on Windows

**Comment by dg0yt** [comment]
> I don't see how a fatal error would help

It would quickly stop CI for triplets you are not interested in.

**Comment by dg0yt** [comment]
> This didn't have the expected result: every port that installs man fails instead of only those without GNUInstallDirs on Windows

I thought this was the desired result: Identify the ports which install man pages. When vcpkg doesn't want to install documentation.

**Comment by dg0yt** [comment]
Instead of patching each port, I would suggest to always redirect `CMAKE_INSTALL_MANDIR` to a designated trash folder which is cleaned at the end of the install.

For a more general approach, how about globbing the items in the root directory of the packages/install dir drive before the build, and check for new items after the build. This would catch a few more problems than just misplaced man pages, independent of write privileges.

**Comment by Thomas1664** [comment]
> It would quickly stop CI for triplets you are not interested in.

Now I understand you. Thanks for the hint!

> For a more general approach, how about globbing the items in the root directory of the packages/install dir drive before the build, and check for new items after the build. This would catch a few more problems than just misplaced man pages, independent of write privileges.

Good idea, but we could only do this in CI (and on Windows) since a user may intentionally create folders in `C:/` during install. I would do it from C++ from the ci command in combination with post build lint.

**Pull request closed without merging**